File: app/src/main/java/uo206367/dspm/miw/dsdmchatt/service/AppDatabase.java
package uo206367.dspm.miw.dsdmchatt.service;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import uo206367.dspm.miw.dsdmchatt.model.User;

@Database(entities = {User.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase INSTANCE;

    public abstract UserDao userDao();

    public static AppDatabase getAppDatabase(Context context) {
        if (INSTANCE == null) {
            INSTANCE =
                    Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "user-database")
                            // allow queries on the main thread.
                            // Don't do this on a real app! See PersistenceBasicSample for an example.
                            .allowMainThreadQueries()
                            .build();
        }
        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }


    private static User addUser(final AppDatabase db, User user){
        db.userDao().insertAll(user);
        return user;
    }

    private static User assertUser(final AppDatabase db, User user){
        db.userDao().findByName(user.getUserName());
    }
}

File: app/src/main/java/uo206367/dspm/miw/dsdmchatt/service/AppDatabase.kt
package uo206367.dspm.miw.dsdmchatt.service

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

import uo206367.dspm.miw.dsdmchatt.model.User

@Database(entities = arrayOf(User::class), version = 1)
abstract class AppDatabase : RoomDatabase() {

    abstract fun userDao(): UserDao

    companion object {

        private var INSTANCE: AppDatabase? = null

        fun getAppDatabase(context: Context): AppDatabase {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "user-database")
                        // allow queries on the main thread.
                        // Don't do this on a real app! See PersistenceBasicSample for an example.
                        .allowMainThreadQueries()
                        .build()
            }
            return INSTANCE
        }

        fun destroyInstance() {
            INSTANCE = null
        }


        private fun addUser(db: AppDatabase, user: User): User {
            db.userDao().insertAll(user)
            return user
        }

        private fun assertUser(db: AppDatabase, user: User): User {
            return db.userDao().findByName(user.userName)
        }
    }
}